### PR TITLE
3347 fixes mednist dataset

### LINKED ISSUE #3347: notebook tests error: ./2d_registration/registration_mednist.ipynb
**Describe the bug**

```
Running ./2d_registration/registration_mednist.ipynb
Checking PEP8 compliance...
Running notebook...
Before:
    "max_epochs = 200\n",
After:
    "max_epochs = 1\n",

Executing:   0%|          | 0/20 [00:00<?, ?cell/s]
Executing:   5%|▌         | 1/20 [00:01<00:24,  1.27s/cell]
Executing:  20%|██        | 4/20 [00:04<00:18,  1.14s/cell]
Executing:  30%|███       | 6/20 [00:06<00:15,  1.14s/cell]
Executing:  40%|████      | 8/20 [00:16<00:30,  2.54s/cell]
Executing:  55%|█████▌    | 11/20 [00:16<00:12,  1.43s/cell]
Executing:  55%|█████▌    | 11/20 [00:17<00:14,  1.57s/cell]
Traceback (most recent call last):
  File "/opt/conda/bin/papermill", line 8, in <module>
    sys.exit(papermill())
  File "/opt/conda/lib/python3.8/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.8/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/papermill/cli.py", line 250, in papermill
    execute_notebook(
  File "/opt/conda/lib/python3.8/site-packages/papermill/execute.py", line 122, in execute_notebook
    raise_for_execution_errors(nb, output_path)
  File "/opt/conda/lib/python3.8/site-packages/papermill/execute.py", line 234, in raise_for_execution_errors
    raise error
papermill.exceptions.PapermillExecutionError: 
---------------------------------------------------------------------------
Exception encountered at "In [6]":
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_2509/823736021.py in <module>
      1 check_ds = Dataset(data=training_datadict, transform=train_transforms)
----> 2 check_loader = DataLoader(check_ds, batch_size=1, shuffle=True)
      3 check_data = first(check_loader)
      4 fixed_image = check_data["fixed_hand"][0][0]
      5 moving_image = check_data["moving_hand"][0][0]

/home/jenkins/agent/workspace/Monai-notebooks/MONAI/monai/data/dataloader.py in __init__(self, dataset, num_workers, **kwargs)
     82             kwargs.update({"worker_init_fn": worker_init_fn})
     83 
---> 84         super().__init__(dataset=dataset, num_workers=num_workers, **kwargs)  # type: ignore[call-overload]

/opt/conda/lib/python3.8/site-packages/torch/utils/data/dataloader.py in __init__(self, dataset, batch_size, shuffle, sampler, batch_sampler, num_workers, collate_fn, pin_memory, drop_last, timeout, worker_init_fn, multiprocessing_context, generator, prefetch_factor, persistent_workers)
    266             else:  # map-style
    267                 if shuffle:
--> 268                     sampler = RandomSampler(dataset, generator=generator)
    269                 else:
    270                     sampler = SequentialSampler(dataset)

/opt/conda/lib/python3.8/site-packages/torch/utils/data/sampler.py in __init__(self, data_source, replacement, num_samples, generator)
    100 
    101         if not isinstance(self.num_samples, int) or self.num_samples <= 0:
--> 102             raise ValueError("num_samples should be a positive integer "
    103                              "value, but got num_samples={}".format(self.num_samples))
    104 

ValueError: num_samples should be a positive integer value, but got num_samples=0
```

introduced since https://github.com/Project-MONAI/MONAI/pull/3332

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3347

### Description
fixes the issue with `iterdir()` when `root_dir` is a relative path

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/utils.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ def extractall(
         cache_dir = Path(output_dir, _basename(filepath).split(".")[0])
     else:
         cache_dir = Path(output_dir)
-    if cache_dir.exists() and len(list(cache_dir.iterdir())) > 0:
+    if cache_dir.exists() and next(cache_dir.iterdir(), None) is not None:
```
Comment (Nic-Ma) [inline monai/apps/utils.py:244]: Maybe we can also use `monai.utils.first()` here?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.

Thanks.

File: monai/apps/utils.py (hunk 1 above)
Comment (wyli) [reply monai/apps/utils.py:244]: May have a circular import here

**Pull request merged**